funcframework/framework_test.go

```diff
@@ -632,6 +632,17 @@ func TestDeclarativeFunctionCloudEvent(t *testing.T) {
 	}
 }
 
+func TestDeclarativeFunctionWithoutRegister(t *testing.T) {
```

Comment:
[nit] This test name is a little misleading, since we should for all functions if we can't find the function target

We use "declarative" to mean fn.go has:
```go
func init() {
    functions.HTTP("MyFunc", myFunc)
}

func myFunc(...) {}
```

vs. "non-declarative"


```go
// Missing init and API call that declares the function target to functions framework

func MyFunc(...){}
```

Both of these function files would be valid with `FUNCTION_TARGET=MyFunc`, and we should error in both cases if `FUNCTION_TARGET` is anything else.